gt/communication/ghex_comm.cpp

```diff
@@ -0,0 +1,59 @@
+#include "./ghex_comm.hpp"
+
+#include <ghex/glue/gridtools/field.hpp>
+
+namespace communication {
+
+namespace ghex_comm {
+
+
+std::function<void(storage_t &)>
+comm_halo_exchanger(grid const &grid, storage_t::storage_info_t const &sinfo)
+{
+    static std::unique_ptr<typename grid::comm_obj_type> comm_obj;
+    if (!comm_obj)
+    {
+        comm_obj.reset( new typename grid::comm_obj_type{::gridtools::ghex::make_communication_object<typename grid::patterns_type>()} );
+    }
+    auto co_ptr = comm_obj.get();
+    auto patterns_ptr = &grid.patterns();
+
+    using domain_id_type    = typename local_domain::domain_id_type;
+    using arch_t            = typename ::gridtools::ghex::_impl::get_arch<storage_t>::type;
+    using value_t           = typename storage_t::data_t;
+    using layout_t          = typename storage_t::storage_info_t::layout_t;
+    using integer_seq       = typename ::gridtools::_impl::get_unmasked_layout_map<layout_t>::integer_seq;
```

Comment:
ghex is part of gridtools universe, and it will remain in `gridtools::ghex`. I'll check whether I leak into `gridtools::_impl`, but I don't think I do, just using internals from gridtools.

Reply:
fixed